G1ANT.Addon.UI/ExtensionMethods/AutomationExtensions.cs

```diff
@@ -20,5 +20,24 @@ public static string GetText(this AutomationElement element)
             }
             return element.Properties.Name;
         }
+
+
+        public static AutomationElement GetParentElementClosestToDesktopElement(this AutomationElement element)
+        {
+            var desktop = AutomationSingleton.Automation.GetDesktop();
+            var currentElement = element;
+            while (!desktop.Equals(currentElement.Parent))
+            {
+                if (currentElement.Parent == null)
```

Comment:
If only element without parent is desktop then this condition is never met. Consider checking it once before loop

Reply:
true, good catch!